src/lib.rs

```diff
@@ -66,6 +68,11 @@ const LOGGING_ENV: &str = "SCCACHE_LOG";
 
 pub fn main() {
     init_logging();
+
+    if !check_valid_port_available(true) {
+        std::process::exit(1);
+    }
```

Comment:
Instead of checking if the port is available at the very beginning of the application, can we instead catch error (probably the `10013` one) when we attempt to allocate the port? At that point, we can recommend setting/changing `SCCACHE_SERVER_PORT`.

The downsides with the current approach are:
* You need to infer from the args whether or not a port will be allocated here, which is unfortunate. We're parsing args twice now :(
* There's a potential race condition (what if the port is available here, but then something else steals it before `sccache` actually does its real port allocation?
* We can probably avoid the `PartialEq` implementations, perhaps.
* We'll definitely be able to remove `get-port`, I believe.